desktop/core/src/desktop/js/catalog/optimizer/SqlAnalyzer.ts

```diff
@@ -98,6 +93,21 @@ export default class SqlAnalyzer implements Optimizer {
     });
   }
 
+  async checkMissingLimit(statement: string, dialect: string): Promise<boolean> {
+    const autocompleter = await sqlParserRepository.getAutocompleteParser(dialect);
+    const parsedStatement = autocompleter.parseSql(statement + ' ', '');
+
+    return (
+      parsedStatement.locations.some(
+        location => location.type === 'statementType' && location.identifier === 'SELECT'
+      ) &&
+      parsedStatement.locations.some(location => location.type === 'table') &&
+      parsedStatement.locations.some(location => {
+        return location.type === 'limitClause' && location.missing;
```

Comment:
Nit: Skip the return for consistency, i.e.

location => location.type === 'limitClause' && location.missing

Reply:
I see, let me fix those